internal/store/store.go

```diff
@@ -540,6 +548,35 @@ func (bs *BlockStore) Close() error {
 	return bs.db.Close()
 }
 
+// Rollback rollbacks the latest block from BlockStore.
+func (bs *BlockStore) Rollback() error {
```

Comment:
How do you intend on using this if the store's in `internal`?

Reply:
it wasn't `internal` >2 days ago :thinking:  https://github.com/cometbft/cometbft/commit/10fa8d1016d98f8395be47cd2a9c40f2ae03e7bd#diff-b26d414d0eb20deebca0d2b785bd3e1090aa1a7a182e7dc30640d68c53cb40da

Ideas?

Reply:
How were you intending on using it in the first place?

See ADR 109 (#1484) for context on the internalization of a number of our public APIs.

Reply:
Technically, if you access the block store via the node's `BlockStore` method you should still be able to access the `Rollback` function.

We cannot, however, guarantee the stability of Go APIs that are defined within `internal`.

Reply:
> How were you intending on using it in the first place?

I've described that in a step-by-step guide here https://gist.github.com/andy108369/da1279257c018be6370310c990c25738

Reply:
What'd help is if we could understand where you want to call that function from. Would it be called from within Comet's own rollback command, within this SDK call here? https://github.com/cosmos/cosmos-sdk/blob/7799bba7bc889288607576aa11655b5fc31a2da9/server/rollback.go#L35

Reply:
I believe within the SDK call based on the step-by-step guide shared prior.
Or whatever makes it easier to just run `akash rollback --delete-pending-block` and make sure it deletes it. :-)

Reply:
Is there any reason not to modify the publicly exported `RollbackState` function in Comet to take a `deletePendingBlock` parameter? https://github.com/cometbft/cometbft/blob/ee99bf534a35f56f0e5ad463d93bbe1d796bf9f0/cmd/cometbft/commands/rollback.go#L54

Reply:
If you want to do this in a non-breaking way (in terms of the public Go API), simply create a new publicly exported function there called `RollbackStateAndPendingBlock` that calls `RollbackState` and deletes the pending block.

Reply:
Thank you @thanethomson ! I'll try it.

Reply:
Hey @arno01, just checking in here to see whether you're going to implement this?